src/components/datepicker/datePicker.js

```diff
@@ -574,15 +579,24 @@
   /** Close the floating calendar pane. */
   DatePickerCtrl.prototype.closeCalendarPane = function() {
     if (this.isCalendarOpen) {
-      this.detachCalendarPane();
-      this.isCalendarOpen = false;
-      this.calendarPaneOpenedFrom.focus();
-      this.calendarPaneOpenedFrom = null;
+      var self = this;
 
-      this.ngModelCtrl.$setTouched();
+      if (self.calendarPaneOpenedFrom) {
+        self.calendarPaneOpenedFrom.focus();
+        self.calendarPaneOpenedFrom = null;
+      }
 
-      this.documentElement.off('click touchstart', this.bodyClickHandler);
-      window.removeEventListener('resize', this.windowResizeHandler);
+      // Ensures that all focus events have fired before detaching
+      // the calendar. Prevents the calendar from reopening immediately
+      // in IE when md-open-on-focus is set.
+      this.$mdUtil.nextTick(function() {
```

Comment:
Wouldn't it make sense to only run the detach-block in the next tick, if the `md-open-on-focus` attribute is set to true? Otherwise you should be able to run this immediately.